src/scripts/main.js

```
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (!element) {
+      reject(new Error(`No element - ${element}`));
+      return;
+    }
```

Validation and fail-fast behavior (critical):

Currently the falsy-element check and rejection happens inside the Promise executor. Prefer validating before creating the Promise and failing fast by throwing synchronously. Also verify the object supports addEventListener (e.g. `typeof element.addEventListener === 'function'`).

Why: synchronous validation surfaces errors immediately to callers and avoids leaving promises rejected/pending in unexpected ways. Place this check at the top of `waitFor` so you don't create a promise when the input is invalid.

Suggested approach (conceptual):

- Before creating the Promise, do:

```javascript
if (!element || typeof element.addEventListener !== 'function') {
  throw new TypeError('Invalid element passed to waitFor');
}
```

Then create `new Promise((resolve) => { ... })` without `reject`.